lua/lackluster/init.lua

```diff
@@ -73,134 +73,134 @@ local USER_CONFIG = nil
-        t.syntax = t.syntax_mint
-        vim.g.colors_name = "lackluster-mint"
-    end
-
-    -- expieramental themes
-    if opt.theme == "dark" then
-        t.syntax = t.syntax_dark
-        vim.g.colors_name = "lackluster-dark"
-    end
-
-    if opt.theme == "night" then
-        t.syntax = t.syntax_night
-        vim.g.colors_name = "lackluster-night"
-    end
-
-    t.syntax = vim.tbl_extend('force', t.syntax, t.syntax_tweak)
-
-    local dedup_set = {}
-    local highlight_group_list = highlight(t, color)
-
-    for _, highlight_group in ipairs(highlight_group_list) do
-        local highlight_spec_list = highlight_group.highlight
-        local is_plugin_enabled = not USER_CONFIG.disable_plugin[highlight_group.plugin_name]
-
-        if highlight_group.dont_skip or is_plugin_enabled then
-            for _, hl_spec in ipairs(highlight_spec_list) do
-                local hl_name = hl_spec.name
-                if dedup_set[hl_name] then
-                    vim.notify("error: duplicate hi_spec :: " .. hl_name, vim.log.levels.ERROR)
-                else
-                    dedup_set[hl_name] = true
-                    hl_spec.name = nil -- must set to nil so that nvim_set_hl doesn't freq out
-                    ---@diagnostic disable-next-line: param-type-mismatch
-                    vim.api.nvim_set_hl(0, hl_name, hl_spec)
-                end
-            end
-        end
-    end
+	opt = opt or {}
+
+	if USER_CONFIG == nil then
+		USER_CONFIG = vim.tbl_deep_extend("force", {}, default_config)
+	end
+
+	local t = theme
+	t.syntax = t.syntax_default
+	vim.o.termguicolors = true
+	vim.g.colors_name = "lackluster"
+
+	-- official themes
+	if opt.theme == "hack" then
+		t.syntax = t.syntax_hack
+		vim.g.colors_name = "lackluster-hack"
+	end
+
+	if opt.theme == "mint" then
+		t.syntax = t.syntax_mint
+		vim.g.colors_name = "lackluster-mint"
+	end
+
+	-- expieramental themes
+	if opt.theme == "dark" then
+		t.syntax = t.syntax_dark
+		vim.g.colors_name = "lackluster-dark"
+	end
+
+	if opt.theme == "night" then
+		t.syntax = t.syntax_night
+		vim.g.colors_name = "lackluster-night"
+	end
+
+	t.syntax = vim.tbl_extend("force", t.syntax, t.syntax_tweak)
+
+	local dedup_set = {}
+	local highlight_group_list = highlight(t, color)
+
+	for _, highlight_group in ipairs(highlight_group_list) do
+		local highlight_spec_list = highlight_group.highlight
+		local is_plugin_enabled = not USER_CONFIG.disable_plugin[highlight_group.plugin_name]
+
+		if highlight_group.dont_skip or is_plugin_enabled then
+			for _, hl_spec in ipairs(highlight_spec_list) do
+				local hl_name = hl_spec.name
+				if dedup_set[hl_name] then
+					vim.notify("error: duplicate hi_spec :: " .. hl_name, vim.log.levels.ERROR)
+				else
+					dedup_set[hl_name] = true
+					hl_spec.name = nil -- must set to nil so that nvim_set_hl doesn't freq out
+					---@diagnostic disable-next-line: param-type-mismatch
+					vim.api.nvim_set_hl(0, hl_name, hl_spec)
```

Comment:
In your bug report #44 it looks like the affending error is on line 199 but I dont see any changes that would affect this call to `nvim_set_hl`

what exactly did you change that is supposed to fix this issue?